### PR TITLE
Issue3

### LINKED ISSUE #3: Image Index
#### As a user I want the homepage to display all saved images.

__Story__:

Now that we are saving images for our users, we want them to be able to view
the list of images in the system. You might be thinking, "what if my users only
want to view images of cats", or "what if my users only want to see the images
they uploaded". Those are great questions. Maybe you will have the opportunity
to add those features in a future story. For now, we will keep it simple and
quickly deliver a little more value to our users.

__Acceptance criteria__:
- [ ] New images that are added show up on the homepage.
- [ ] These images are persisted if the browser is closed or even if the
  server is restarted.
- [ ] Images are not displayed wider than 400px.
- [ ] Newest images appear first.

__Dependencies__:
- Save Image Link

### PR DESCRIPTION
**Pull request by tmcswain**
Closes #3

### CODE REVIEW

**Comment by tmcswain** [comment]
[Issue 3: Image Index](https://trello.com/c/EtLhJCDi/276-issue-3-image-index)

File: test/controllers/application_controller_test.rb [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+      Image.create!(url: url)
+    end
+
+    num_images = Image.count
+
+    get root_url
+
+    assert_response :success
+    assert_select 'img' do |images|
+      images.each_with_index do |img, i|
```
Comment (mohanck) [inline test/controllers/application_controller_test.rb]: 👍

File: test/controllers/application_controller_test.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 require 'test_helper'
 class ApplicationControllerTest < ActionDispatch::IntegrationTest
-  def test_home__displays_h1
+  def test_home__displays_h1_and_contains_link
     get root_url
     assert_response :success
-    assert_select 'h1', 'Hi! Welcome!!!'
+    assert_select 'h1', 'Welcome!'
+    assert_select 'a', 'New Image', href: new_image_path
```
Comment (natalieschauser) [inline test/controllers/application_controller_test.rb]: I think that `assert_select` will ignore the `href` condition in this case. I tried running the test with
`assert_select 'a', 'New Image', href: 'some random url'` and it still passed. 

I think the  syntax should be something more like:
`assert_select "a[href=?]", "/desired_path/1"`

File: test/controllers/application_controller_test.rb [hunk 3]
```diff
@@ -1,9 +1,36 @@
...
+  def test_home
     get root_url
+
     assert_response :success
     assert_select 'h1', 'Welcome!'
     assert_select 'a', 'New Image', href: new_image_path
+    assert_select 'img', 0
+  end
+
+  def test_home_displays_all_images_in_order
```
Comment (natalieschauser) [inline test/controllers/application_controller_test.rb]: This is minor, but usually the way we write test names is as follows:
`test_action_to_test__description` (notice the two underscores between the action to test and the name. 

Maybe we can re-write this test name to be as follows:
`test_home__displays_all_images_in_order`

File: test/controllers/application_controller_test.rb [hunk 4]
```diff
@@ -1,9 +1,36 @@
...
+    end
+
+    num_images = Image.count
+
+    get root_url
+
+    assert_response :success
+    assert_select 'img' do |images|
+      images.each_with_index do |img, i|
+        assert_equal urls[num_images - i - 1], img.attr('src')
```
Comment (natalieschauser) [inline test/controllers/application_controller_test.rb]: Again minor, but this is pretty hard to read. Maybe we can create an `expected_urls` array that is in the correct order and then just index into that?

**Pull request merged**